framework/db/Query.php

```diff
@@ -470,13 +470,20 @@ public function getTablesUsedInFrom()
             $tableNames = $this->from;
         } elseif (is_string($this->from)) {
             $tableNames = preg_split('/\s*,\s*/', trim($this->from), -1, PREG_SPLIT_NO_EMPTY);
+        } elseif ($this->from instanceof Expression) {
+            return (string) $this->from;
```

Comment:
Doesn't looks correct. `getTablesUsedInFrom()` return array of strings, not a string.